drivers/potentiometer/ad5293/ad5293.h

```diff
@@ -0,0 +1,149 @@
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#ifndef _ad5293_H_
+#define _ad5293_H_
+/*****************************************************************************/
+/****************************** Include Files ********************************/
+/*****************************************************************************/
+#include <stdint.h>
+#include "no_os_gpio.h"
+#include "no_os_spi.h"
+#include "no_os_util.h"
+
+/* SPI Read/Write commands */
+#define CMD_NOP        0b0000
+#define CMD_W_WIPER    0b0001
+#define CMD_R_WIPER    0b0010
+#define CMD_RESET      0b0100
+#define CMD_W_CTRL     0b0110
+#define CMD_R_CTRL     0b0111
+#define CMD_PD         0b1000
+#define CMD_SDO_Z      0x8001
+
+/* Bit mask*/
+#define CALI_MSK       NO_OS_BIT(2)
+#define PROTECT_MSK    NO_OS_BIT(1)
+
+/* calibration mode */
+enum calibration_mode_t {
+	CALI_PERFORMANCE,
+	CALI_NORMAL
+};
+
+/* protect mode */
+enum protect_mode_t {
+	PROTECT_LOCK,
+	PROTECT_UNLOCK // enable update wiper position through digital interface
+};
+
+/* shutdown mode */
+enum shutdown_t {
+	NORMAL,
+	SHUTDOWN
+};
+
+/* operation mode */
+enum operation_mode_t {
+	BUS,
+	DAISY_CHAIN
+};
+
+/* SDO mode */
+enum SDO_mode_t {
+	SDO_DISENABLE,
+	SDO_ENABLE
+};
+
+/* SPI frame */
+#define AD5293_W_CTRL_FRAME(pm,cali)     (uint16_t)((CMD_W_CTRL<<10)|(pm<<1)|(cali<<2))
+#define AD5293_R_CTRL_FRAME              (uint16_t)CMD_R_CTRL<<10
+#define AD5293_W_WIPER_FRAME(data)       (uint16_t)((CMD_W_WIPER<<10)|data)
+#define AD5293_R_WIPER_FRAME             (uint16_t)CMD_R_WIPER<<10
+#define SHUTDOWN_FRAME(sd)               (uint16_t)((CMD_PD<<10)|sd)
+#define RESET_FRAME                      (uint16_t)CMD_RESET<<10
+
+/**********************************struct*****************************************/
+struct ad5293_chip_info {
+	enum calibration_mode_t cali;
+	uint16_t Wiper_value; //10bit
+};
+
+struct ad5293_dev {
+	/* SPI */
+	struct no_os_spi_desc	*spi_desc;
+	/* GPIO */
+	struct no_os_gpio_desc	*gpio_reset;
+	/* number of chips */
+	uint16_t chip_num;
+	/* pointer of chip struct  */
+	struct ad5293_chip_info* chip; //point to chip 0
```

Comment:
please document better the fact that this is an array and which instance it's used (daisy chain?)